src/channel.py

```diff
@@ -1,19 +1,27 @@
 import sys
 import traceback
-from typing import Any
+from typing import Any, Tuple
 import pickle
-from lib import Branch
+from lib import Branch, type_to_str
+from session import Session
 from sessiontype import *
-from enum import Enum
 import socket
+import statemachine
+from statemachine import Action
+from check import typecheck_file
+
 
-T = TypeVar('T')
 
+T = TypeVar('T')
 
 class Channel(Generic[T]):
```

Comment:
remove the 'local mode' concept. We dont need threading

Reply:
That, you will do, young padawan